user_video.py

```diff
@@ -55,4 +76,38 @@ def get_immediate_frame(self):
         print(f"processed_latest_inboard_frame {processed_latest_inboard_frame}")
         return processed_latest_inboard_frame
     
+    def get_predictions(self, image_path):
+        """
+        Return the results of text detection, text recognition and translation for the Input Image.
+        """
+        results = []
+
+        print(image_path)
+        game, lang, number = image_path.stem.split('_')
+        img = Image.open(image_path)
+        img = self.video_stream.preprocess_image(img, crop_y_coordinate=self.crop_height)
+        
+        image = self.video_stream.textDetector.preprocess_image(image=img)
+        if not self.video_stream.textDetector.has_text(image):
+            return {
+                "image": str(image_path),
+                "text_detected": False,
+                "recognized_text": '',
+                "annotations": [],
+                "translation": ''
+            }
+        
+        if lang == 'EN':
+            closest_match, previous_image, highlighted_image, annotations, translation = self.frameProcessor_en.run_image(img, translate=self.video_stream.background_task_args.get("translate", None), enable_cache=self.video_stream.background_task_args.get("enable_cache", False))
```

Comment:
can we keep a single instance to a single language and we don't need to do this frameprocess_en / frameprocessor_jp

Reply:
I've changed the code for this.